File: src/main/java/com/shnok/serverpackets/SystemMessagePacket.java
package com.shnok.serverpackets;

public class SystemMessagePacket extends ServerPacket {

    public enum MessageType {
        USER_LOGGED_IN,
        USER_LOGGED_OFF
    }

    public MessageType _type;

    public SystemMessagePacket(MessageType type) {
        super((byte)0x03);
        _type = type;
    }

    public SystemMessagePacket(MessageType type, String value) {
        super((byte)0x03);
        _type = type;

        writeB((byte)type.ordinal());
        writeS(value);

        buildPacket();
    }
}

File: Assets/Scripts/Network/ServerPackets/SystemMessagePacket.cs
using UnityEngine;
using System;

public enum MessageType {
    USER_LOGGED_IN,
    USER_LOGGED_OFF
}

public class SystemMessagePacket : ServerPacket {

    private MessageType type;
    private SystemMessage message;

    public SystemMessagePacket(){}
    public SystemMessagePacket(byte[] d) : base(d) {
        Parse();
    }
    
    public override void Parse() {    
        try {
            type = (MessageType)ReadB();

            switch (type) {
                case MessageType.USER_LOGGED_IN:
                    message = new MessageLoggedIn(ReadS());
                    break;
                case MessageType.USER_LOGGED_OFF:
                    message = new MessageLoggedOut(ReadS());
                    break;
            }            
        } catch(Exception e) {
            Debug.Log(e);
        }
    }

    public SystemMessage GetMessage() {
        return message;
    }
}